### PR TITLE
Catch member errors

### PR DESCRIPTION
**Pull request by ajuvercr**
* Actually use concurrent requests
* cleanup logging a bit
* catch member extract errors

### CODE REVIEW

File: lib/memberManager.ts [hunk 1]
```diff
@@ -149,16 +164,23 @@ export class Manager {
...
+            if (member) {
+              notifier.extracted(member, state);
+            }
+            return member;
+          })
+          .catch((ex) => {
+            notifier.error(
+              { error: ex, type: "extract", memberId: member },
+              state,
+            );
```
Comment (pietercolpaert) [inline lib/memberManager.ts:182]: Add `return null`?

File: lib/client.ts [hunk 2]
```diff
@@ -430,7 +429,7 @@ export class Client {
       cancel: async () => {
         this.stateFactory.write();
         console.log("Canceled");
```
Comment (kkostov) [inline lib/client.ts:431]: Can this trace be removed or switched to the `logger`?

File: lib/memberManager.ts [hunk 3]
```diff
@@ -66,41 +73,50 @@ export class Manager {
...
 
+    // let the extractor do it's thing
+    return await this.extractor.extract(data, member);
+  }
+
+  private async extractMember(
+    member: Term,
+    data: RdfStore,
+  ): Promise<Member | undefined> {
+    const quads: Quad[] = await this.extractMemberQuads(member, data);
```
Comment (kkostov) [inline lib/memberManager.ts:118]: I think the `extractMemberQuads` as well as the `extract` method coming from the cbd shape package can still reject their promise without it being handled.

File: lib/memberManager.ts [hunk 4]
```diff
@@ -149,16 +164,23 @@ export class Manager {
...
+          })
+          .catch((ex) => {
+            notifier.error(
+              { error: ex, type: "extract", memberId: member },
+              state,
+            );
+          });
 
         promises.push(promise);
       }
```
Comment (kkostov) [inline lib/memberManager.ts:186]: The https://github.com/rdf-connect/ldes-client/pull/24/files#diff-da7e57eee5774b6f1726c34a35836992bfb7e13ff97057f20d9699901b542171L167 Promises.all() should have a .catch()

**Comment by pietercolpaert** [comment]
The concurrent requests now also refetch concurrently, which of course then keeps resulting in problems:

```bash
pieter@pieter-Latitude-7430:~/Projects/LDES-client$ export DEBUG=*;ts-node bin/cli.ts https://marineregions.org/feed.ttl --retry-count 0
...
  fetch:retry Retry https://www.marineregions.org/feed.ttl?page=2023-10-02T11%3A00%3A00Z%2F2023-10-02T12%3A00%3A00Z 114/0 +5ms
  fetch:retry Retry http://marineregions.org/mrgid/7494?t=1704805889 156/0 +1ms
  fetch:retry Retry http://marineregions.org/mrgid/7496?t=1704805889 153/0 +5ms
  ...
```

I’d also look rename the `--retry-count` parameter to `--max-retries`

**Comment by pietercolpaert** [comment]
I sit also possible there’s no exponential back-off anymore, and it just retires as fast as possible?

**Comment by pietercolpaert** [comment]
I think this is fixed now - can you check?

**Pull request merged**